consensus/hotstuff/cruisectl/block_time_controller.go

```diff
@@ -450,17 +430,25 @@ func (ctl *BlockTimeController) measureViewDuration(tb TimedBlock) error {
 
 // processEpochExtended processes the EpochExtended protocol data.
 // Whenever we encounter an epoch extension, we:
-//   - set epochFallbackTriggered to true if this is the first extension encountered.
 //   - update the curr epoch final view and target end time with the extension data.
 //
 // No errors are expected during normal operation.
-func (ctl *BlockTimeController) processEpochExtended(_ uint64, _ *flow.Header, extension flow.EpochExtension) error {
-	if !ctl.epochFallbackTriggered {
-		// set epochFallbackTriggered to true when we encounter the first extension
-		ctl.epochFallbackTriggered = true
+func (ctl *BlockTimeController) processEpochExtended(header *flow.Header, extension flow.EpochExtension) error {
```

Comment:
```suggestion
func (ctl *BlockTimeController) processEpochExtended(first *flow.Header) error {
	finalView, err := ctl.state.AtHeight(header.Height).Epochs().Current().FinalView()
	if err != nil {
		return fmt.Errorf("failed to retrieve final view at block id %s", header.ID())
	}
	targetEndTime, err := ctl.state.AtHeight(header.Height).Epochs().Current().TargetEndTime()
	if err != nil {
		return fmt.Errorf("failed to retrieve target end time at block id %s", header.ID())
	}
```

A few small suggestions here:
- since we need to query the protocol state anyway, get both the final view and the target end time from querying the protocol state to eliminate the possibility that the two parameters are inconsistent
- use the `first` parameter name to match the interface definition and the other event handler function in this component
- use AtHeight (save an ID computation and match the other event handler function in this component)